src/builder.rs

```diff
@@ -506,4 +506,112 @@ impl<'fbb> Builder<'fbb> {
+            || (first_projected.distance_along_curve == last_projected.distance_along_curve
+                && first_projected.offset.abs() > last_projected.offset.abs())
+        {
+            self.reverse(traversal_index);
+        }
+        Ok(())
+    }
+}
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+    use geo::{coord, point};
+
+    fn build_traversal(builder: &mut Builder) -> usize {
+        let s1 = builder.add_segment("s1", &[coord! {x: 0., y: 0.}, coord! {x:1., y: 0.}], 0, 1);
+        let s2 = builder.add_segment("s2", &[coord! {x: 1., y: 0.}, coord! {x:2., y: 0.}], 1, 2);
+        let sot1 = super::SegmentOfTraversal {
+            segment_index: s1,
+            reversed: false,
+        };
+        let sot2 = super::SegmentOfTraversal {
+            segment_index: s2,
+            reversed: false,
+        };
+        builder.add_traversal("traversal", &[sot1, sot2])
+    }
+
+    #[test]
+    fn traversal_nodes_order() {
+        // Nominal case
+        let mut b = Builder::new();
+        let traversal = build_traversal(&mut b);
+        assert_eq!(b.nodes_of_traversal[traversal], [0, 1, 2]);
+
+        // One reversed edge
+        let s1 = b.add_segment("s1", &[coord! {x: 0., y: 0.}, coord! {x:1., y: 0.}], 10, 11);
+        let s2 = b.add_segment("s2", &[coord! {x: 1., y: 0.}, coord! {x:2., y: 0.}], 10, 12);
+        let sot1 = super::SegmentOfTraversal {
+            segment_index: s1,
+            reversed: true,
+        };
+        let sot2 = super::SegmentOfTraversal {
+            segment_index: s2,
+            reversed: false,
+        };
+        let traversal = b.add_traversal("traversal", &[sot1, sot2]);
+        assert_eq!(b.nodes_of_traversal[traversal], [11, 10, 12]);
+    }
+
+    #[test]
+    fn orientation_on_curve() {
+        let mut b = Builder::new();
+        let traversal = build_traversal(&mut b);
+
+        let point_a = point! {x: 1., y: 1.};
+        let point_b = point! {x: 2., y: -1.};
+
+        b.orient_along_points(traversal, point_a, point_b).unwrap();
+        assert_eq!(b.temp_traversal[0].segments[0].segment_index, 0);
+        assert_eq!(b.nodes_of_traversal[0][0], 0);
+
+        b.orient_along_points(traversal, point_b, point_a).unwrap();
+        assert_eq!(b.temp_traversal[0].segments[0].segment_index, 1);
+        assert_eq!(b.nodes_of_traversal[0][0], 2);
+    }
+
+    #[test]
+    fn orientation_before_curve() {
+        let mut b = Builder::new();
+        let traversal = build_traversal(&mut b);
+
+        // a-b---[the curve]
+        let point_a = point! {x: -1., y: 0.};
+        let point_b = point! {x: -2., y: 0.};
+        b.orient_along_points(traversal, point_a, point_b).unwrap();
+        assert_eq!(b.temp_traversal[traversal].segments[0].segment_index, 0);
+        b.orient_along_points(traversal, point_b, point_a).unwrap();
+        assert_eq!(b.temp_traversal[traversal].segments[0].segment_index, 1);
```

Comment:
Why not check b.nodes_of_traversal here ?

Reply:
In the first test we want to test that everything has been reversed

here we want to test some edge case behavior when using points that project to an extremity of the curve

Reply:
Okay, fair enough